src/components/common/NetworkSelector/NetworkMultiSelector.tsx

```diff
@@ -0,0 +1,76 @@
+import useChains from '@/hooks/useChains'
+import { type ReactElement, useState } from 'react'
+import { Checkbox, Autocomplete, TextField, Chip } from '@mui/material'
+import type { ChainInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import ChainIndicator from '../ChainIndicator'
+import { useRouter } from 'next/router'
+import useWallet from '@/hooks/wallets/useWallet'
+import { getNetworkLink } from '.'
+import css from './styles.module.css'
+
+const NetworkMultiSelector = ({
+  selectedNetworks,
+  setSelectedNetworks,
+}: {
+  selectedNetworks: ChainInfo[]
+  setSelectedNetworks: (selectedChains: ChainInfo[]) => void
```

Comment:
We could get rid of those props if this component uses `useFormContext` to pass the values to the react-hook-form.